tests/e2e/test_rosacli_cluster.go

```diff
@@ -2248,3 +2248,206 @@ var _ = Describe("Reusing opeartor prefix and oidc config to create clsuter", la
+		BeforeEach(func() {
+
+			By("Init the client")
+			rosaClient = rosacli.NewClient()
+			clusterService = rosaClient.Cluster
+			ocmResourceService = rosaClient.OCMResource
+
+			By("Get AWS account id")
+			rosaClient.Runner.JsonFormat()
+			whoamiOutput, err := ocmResourceService.Whoami()
+			Expect(err).To(BeNil())
+			rosaClient.Runner.UnsetFormat()
+			whoamiData := ocmResourceService.ReflectAccountsInfo(whoamiOutput)
+			AWSAccountID = whoamiData.AWSAccountID
+
+			By("Prepare custom profile")
+			customProfile = &profilehandler.Profile{
+				ClusterConfig: &profilehandler.ClusterConfig{
+					HCP:           true,
+					MultiAZ:       true,
+					STS:           true,
+					OIDCConfig:    "managed",
+					NetworkingSet: true,
+					BYOVPC:        true,
+					Zones:         "",
+				},
+				AccountRoleConfig: &profilehandler.AccountRoleConfig{
+					Path:               "",
+					PermissionBoundary: "",
+				},
+				Version:      "latest",
+				ChannelGroup: "candidate",
+				Region:       "us-west-2",
+			}
+			customProfile.NamePrefix = constants.DefaultNamePrefix
+
+		})
+
+		AfterEach(func() {
+			By("Delete cluster")
+			rosaClient.Runner.UnsetArgs()
+			_, err := clusterService.DeleteCluster(clusterID, "-y")
+			Expect(err).To(BeNil())
+
+			rosaClient.Runner.UnsetArgs()
+			err = clusterService.WaitClusterDeleted(clusterID, 3, 30)
+			Expect(err).To(BeNil())
+
+			By("Delete operator-roles")
+			_, err = ocmResourceService.DeleteOperatorRoles(
+				"-c", clusterID,
+				"--mode", "auto",
+				"-y",
+			)
+			Expect(err).To(BeNil())
+
+			By("Clean resource")
+			errs := profilehandler.DestroyResourceByProfile(customProfile, rosaClient)
+			Expect(len(errs)).To(Equal(0))
+		})
+
+		It("Check the output of the STS cluster creation with new oidc flow - [id:75925]",
+			labels.Medium, labels.Runtime.Day1Supplemental,
+			func() {
+				By("Create hcp cluster in auto mode")
+				testingClusterName = "rosa75925"
+				testOperatorRolePrefix := "ros75925opp"
+				flags, err := profilehandler.GenerateClusterCreateFlags(customProfile, rosaClient)
+				Expect(err).ToNot(HaveOccurred())
+
+				command := "rosa create cluster --cluster-name " + testingClusterName + " " + strings.Join(flags, " ")
+				rosalCommand := config.GenerateCommand(command)
+				rosalCommand.ReplaceFlagValue(map[string]string{
+					"--operator-roles-prefix": testOperatorRolePrefix,
+				})
+
+				rosalCommand.AddFlags("--mode", "auto")
+				rosalCommand.AddFlags("--billing-account", AWSAccountID)
+				stdout, err := rosaClient.Runner.RunCMD(strings.Split(rosalCommand.GetFullCommand(), " "))
+				Expect(err).To(BeNil())
+				Expect(stdout.String()).To(ContainSubstring("Attached trust policy"))
+
+				rosaClient.Runner.UnsetArgs()
+				clusterListout, err := clusterService.List()
+				Expect(err).To(BeNil())
+				clusterList, err := clusterService.ReflectClusterList(clusterListout)
+				Expect(err).To(BeNil())
+				clusterID = clusterList.ClusterByName(testingClusterName).ID
```

Comment:
how about adding `Expect(clusterID).ToNot(BeNil()) ` for validation, however i guess the clusterID is being used in afterEach and will be validated......but for debug purpose i think its good if we validate it at case level only...WDYT?

Reply:
This ClusterID will be used for AfterEach and also on some following step like waiting cluster to some status. If the ClusterID is empty, it would be failed there. I updated as your suggestion for better readable.